astropy/config/configuration.py

```diff
@@ -402,7 +403,11 @@ def reload(self):
             The new value loaded from the configuration file.
 
         """
-        self.set(self.defaultvalue)
+        try:
+            del self.value
+        except AttributeError:
+            pass
```

Comment:
Maybe like this instead? Though I don't see how this is equivalent to `self.set(self.defaultvalue)`.

```suggestion
        if hasattr(self, "value"):
            del self.value
```

Reply:
For `self.set(self.defaultvalue)` it's annoying but I don't remember :grimacing: (I really should open PRs faster!)

Reply:
note that the `try/except` form is more robust to parallelism (which *might* matter in the wake of free-threaded Python ?)

Reply:
As long as hitting `except` is rare...